src/basic/parse-util.c

```diff
@@ -589,3 +589,13 @@ int parse_ip_port(const char *s, uint16_t *ret) {
 
         return 0;
 }
+
+int parse_dev(const char *s, dev_t *ret) {
+        unsigned major, minor;
```

Comment:
major() and minor() are macros defined in sysmacros.h, the same place makedev() is defined. I think it's better not to name variables like that and avoid any collision. Maybe just name them "a"/"b" or "x"/"y"...